### PR TITLE
Implement real-world readGPS

### PR DESCRIPTION
**Pull request by nhatch**
I tested this on the hardware; see comment below.

### CODE REVIEW

File: src/gps/read_usb_gps.cpp
Comment (evan3334) [inline src/gps/read_usb_gps.cpp:10]: Should we put these in `Constants.h`?

**Review by evan3334 — COMMENTED** [review]
Looks good to me, I think I just had one comment about whether we should put the constants in `Constants.h`.

**Comment by evan3334** [comment]
Merged #99; we can go ahead and merge this if it's ready.

File: src/gps/read_usb_gps.cpp
Comment (evan3334) [reply src/gps/read_usb_gps.cpp:10]: Also, it might be a better idea to use constexpr for these since they're primitive types.

Comment (nhatch) [reply src/gps/read_usb_gps.cpp:10]: Unless we're planning to use these constants elsewhere in the codebase, I think it's easier to keep the constants here, close to where they're used. I've been meaning to move some of our other constants _out_ of `Constants.h`, actually.

**Comment by nhatch** [comment]
I tried walking around the block with the GPS sensor connected to the Rover program. This is the path the GPS thinks I took:

Screenshot from 2021-05-22 13-19-07

It doesn't really make sense, because I think my block should be north/south axis-aligned. Also there were some `nan` readings which I need to investigate---not sure what caused those. But we're getting data of some kind! I think this is ready for review / merge.

Raw data attached
[gps_data.txt](https://github.com/huskyroboticsteam/PY2020/files/6527070/gps_data.txt)

File: src/gps/read_usb_gps.cpp
Comment (evan3334) [inline src/gps/read_usb_gps.cpp:105]: Does the GPS thread ever get stopped before we exit the program? One issue I ran into with the threading in the camera system was that threads will throw an exception if they're forcefully terminated instead of gracefully stopped. Maybe we should have some `running` variable that's used as the `while` loop condition in the thread, and then have a `stopGPSThread()` function that sets it to false and calls `join()` on the thread just to make sure it stops properly.

**Review by evan3334 — CHANGES_REQUESTED** [review]
Looks good to me, just had a note regarding the GPS thread.

File: src/gps/read_usb_gps.cpp
Comment (nhatch) [reply src/gps/read_usb_gps.cpp:105]: I agree there may be some advantages to explicitly joining this thread. Maybe at some point we will have a program that could call startGPSThread multiple times, so it would need to clean up after itself? However, at least on my computer, it doesn't throw any errors if I interrupt the program to kill it. In this case it might be kinder to future maintainers not to add extra code if it doesn't have some concrete benefit. What do you think?

Comment (evan3334) [reply src/gps/read_usb_gps.cpp:105]: Oh weird. Maybe I was mistaken about the error thing? Regardless, if it works okay then it's probably fine. I agree it's best to keep the code simple for now; if it's a problem later we can cross that bridge when we come to it.

**Review by evan3334 — APPROVED** [review]

**Pull request merged**
